protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -129,16 +141,21 @@ public T parse(InputStream stream) {
           if (stream instanceof KnownLength) {
             int size = stream.available();
             if (size > 0 && size <= GrpcUtil.DEFAULT_MAX_MESSAGE_SIZE) {
-              byte[] buf = new byte[size];
+              // Coded Input stream does not escape, so buf does not escape.
```

Comment:
No, it does escape the scope of the method. Namely to `Parser.parse(CodedInputStream)`. Whether protobuf _retains_ the byte array is a different question. I guess the terminology is off. Not important enough to do another PR though :).